src/components/Dao/DaoProposalsPage.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+import * as css from "./DaoProposalsPage.scss";
+import { Address, IDAOState, Proposal } from "@daostack/arc.js";
+import { BreadcrumbsItem } from "react-breadcrumbs-dynamic";
+import withSubscription, { ISubscriptionProps } from "components/Shared/withSubscription";
+import Loading from "components/Shared/Loading";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { Observable } from "rxjs";
+import ProposalRow from "components/Proposal/ProposalRow";
+import { RouteComponentProps } from "react-router-dom";
+import { first } from "rxjs/operators";
+import { isAddress } from "lib/util";
+import moment from "moment-timezone";
+
+const PAGE_SIZE = 50;
+
+type SubscriptionData = Array<Proposal>;
+type IProps = IExternalProps & ISubscriptionProps<SubscriptionData>;
+type IExternalProps = {
+  daoState: IDAOState;
+  currentAccountAddress: Address;
+} & RouteComponentProps<any>;
+
+type OrderDirection = "asc" | "desc";
+
+
+const proposalsQuery = (dao: IDAOState, skip: number, titleSearch?: string): Observable<Array<Proposal>> => {
+  const filter: any = {
+  };
+
+  let orderDirection: OrderDirection = "asc";
+
+  if (!titleSearch) {
+    if (skip % PAGE_SIZE === 0) {
```

Comment:
Could you please explain this logic? Looks like a potential problem

Reply:
`skip` is the size of the total data (proposals) that we currently show on the UI (the proposals that we've received so far).
if `skip % PAGE_SIZE !== 0` it means that we have received less proposals then the page size in the last query, meaning we finished to fetch all the proposals with `closingAt` greater than `0` and we can change the filter to fetch the rest of the proposals. I do agree that it might be a problem if the last chank of proposals will be at size of `PAGE_SIZE`. It's kind of a hack to determine when we received all the data, and it should be improved to be more robust. @orenyodfat @Razzwan